Fixes to source_writer.cc.

case540 opened the pull request:
- Fix memory leak in source_writer constructor.
- FIx test data having .java extension causing issues with internal
  linters. Changing to .txt extension.
- Fix test data path not being correct internally.

Commit:
Fixes to source_writer.cc.

- Fix memory leak in source_writer constructor.
- FIx test data having .java extension causing issues with internal
  linters. Changing to .txt extension.

case540 commented on tensorflow/java/src/gen/cc/source_writer.cc:
```diff
@@ -23,10 +23,17 @@ namespace tensorflow {
 }
 
+SourceWriter::~SourceWriter() {
+  // Remove empty generic namespace added at start.
```
It seems like PushGenericNamespace and PopGenericNamespace always have to be called in pairs (pushing and popping a GenericNamespace to the stack). But the initial GenericNamespace that is pushed (the comment states "as a simplification") doesn't have a corresponding pop.

@karllessard, do you have any comments about this fix

annarev replied:
This seems to rely on user of SourceWriter always calling PopGenericNamespace whenever they called PushGenericNamespace. Since it is left to the class user, I think it is best to handle the case when is not called.

karllessard replied:
@case540: you are right. In fact, like @annarev was pointing out, I think it would be safer to empty our stack in the destructor to delete any remaining namespace that might have been left 'opened' inadvertently, not only the default one.

case540 replied:
Done. Thanks!

Commit:
Addressed comments.

- Free all un-popped GenericNamespace objects in destructor (not
  just the one garunteed to be there).
- Renamed test file to test.java.snippet.

case540 commented on tensorflow/java/BUILD:
```diff
@@ -312,7 +312,7 @@ tf_cc_test(
     ],
     data = [
-        "src/gen/resources/test.snippet.java",
+        "src/gen/resources/test.snippet.java.txt",
```
Renamed to test.java.snippet. I like that name better as well.

case540 commented:
Updated PR / Addressed comments. ty!
